cmd/oceantv/broadcast_hardware_machine.go

```diff
@@ -238,22 +242,39 @@ func (sm *hardwareStateMachine) handleTimeEvent(t timeEvent) {
 	eventIfStatus := func(e event, status bool) {
 		sm.ctx.camera.publishEventIfStatus(e, status, sm.ctx.cfg.CameraMac, sm.ctx.store, sm.log, sm.ctx.bus.publish)
 	}
-	switch sm.currentState.(type) {
+	switch state := sm.currentState.(type) {
 	case *hardwareStarting:
-		withTimeout := sm.currentState.(stateWithTimeout)
-		if withTimeout.timedOut(t.Time) {
+		if state.timedOut(t.Time) {
 			sm.ctx.bus.publish(hardwareStartFailedEvent{})
 			sm.transition(newHardwareOff())
 			return
 		}
 		eventIfStatus(hardwareStartedEvent{}, true)
 	case *hardwareStopping:
+		if state.timedOut(t.Time) {
+			if state.PreviouslyTimedOut {
+				sm.log("hardwareStopping state timed out for the second time")
+				sm.ctx.bus.publish(hardwareStopFailedEvent{})
+				sm.transition(newHardwareOn())
+				return
+			}
+			sm.log("hardwareStopping state timed out, performing controller off actions")
+			err := extStopController(context.Background(), sm.ctx.cfg, sm.ctx.log)
+			if err != nil {
+				sm.log("could not perform controller off actions: %v", err)
+				sm.ctx.bus.publish(hardwareStopFailedEvent{})
+				sm.transition(newHardwareOn())
+				return
+			}
+			state.PreviouslyTimedOut = true
+			state.reset(5 * time.Minute) // 5 minutes is the initial timeout default, so we just reset it here.
```

Comment:
If this is a default somewhere, is there a constant that this can reference?

Reply:
There isn't. Could make it so though.